### PR TITLE
feat(eigen-client-extra-features): address PR comments (part 4)

### PR DESCRIPTION
**Pull request by gianbelinche**
## What ❔
[`data_availability` full scan](https://github.com/matter-labs/zksync-era/pull/3243#discussion_r1888687629)
[simpler ABI decoding](https://github.com/matter-labs/zksync-era/pull/3243#discussion_r1888803680)
[infalible cast](https://github.com/matter-labs/zksync-era/pull/3243#discussion_r1888781030)
[default disperser_rpc](https://github.com/matter-labs/zksync-era/pull/3243#discussion_r1888672059)
[replace `String` with `Url`](https://github.com/matter-labs/zksync-era/pull/3243#discussion_r1888718196)

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev lint`.

### CODE REVIEW

File: core/node/da_clients/src/eigen/verifier.rs [hunk 1]
```diff
@@ -362,38 +365,16 @@ impl Verifier {
...
 
     fn decode_bytes(&self, encoded: Vec<u8>) -> Result<Vec<u8>, String> {
-        // Ensure the input has at least 64 bytes (offset + length)
-        if encoded.len() < 64 {
-            return Err("Encoded data is too short".to_string());
+        let output_type = [ParamType::Bytes];
+        let tokens: Vec<Token> = ethabi::decode(&output_type, &encoded)
+            .map_err(|_| "Incorrect result on contract call")?;
+        let token = tokens
+            .get(0)
```
Comment (juan518munoz) [inline core/node/da_clients/src/eigen/verifier.rs:372]: clippy suggeestion:
```suggestion
            .first()
```

File: core/node/da_clients/src/eigen/verifier.rs [hunk 2]
```diff
@@ -362,38 +365,16 @@ impl Verifier {
...
+            .ok_or_else(|| "Incorrect result on contract call")?;
```
Comment (juan518munoz) [inline core/node/da_clients/src/eigen/verifier.rs:373]: Another clippy suggestion:
```suggestion
            .ok_or("Incorrect result on contract call")?;
```

File: core/node/da_clients/src/eigen/verifier.rs [hunk 3]
```diff
@@ -103,8 +103,8 @@ impl Verifier {
     pub const G2POINT: &'static str = "g2.point.powerOf2";
     pub const POINT_SIZE: u32 = 32;
 
-    async fn save_point(url: String, point: String) -> Result<(), VerificationError> {
-        let url = Url::parse(&url).map_err(|_| VerificationError::LinkError)?;
+    async fn save_point(url: Url, point: String) -> Result<(), VerificationError> {
+        //let url = Url::parse(&url).map_err(|_| VerificationError::LinkError)?;
```
Comment (juan518munoz) [inline core/node/da_clients/src/eigen/verifier.rs:107]: Remove unused, commented code

**Review by juan518munoz — CHANGES_REQUESTED** [review]

**Review by juan518munoz — APPROVED** [review]

File: core/node/da_clients/src/eigen/verifier.rs [hunk 4]
```diff
@@ -362,38 +364,14 @@ impl Verifier {
     }
 
     fn decode_bytes(&self, encoded: Vec<u8>) -> Result<Vec<u8>, String> {
```
[the remark this answers is not in the document]
Comment (gianbelinche) [reply core/node/da_clients/src/eigen/verifier.rs]: Done!

**Review by juanbono — APPROVED** [review]

**Pull request merged**